tools/batchTestGenerator/internal/githubGenerator.go

```diff
@@ -42,10 +44,23 @@ func GithubGenerator(config RunConfig) error {
 		return fmt.Errorf("failed to marshal batch values object: %w", err)
 	}
 
-	fmt.Printf(`::set-output name=batch-keys::%s`, githubBatchKeys)
-	fmt.Printf("\n")
-	fmt.Printf(`::set-output name=batch-values::%s`, githubBatchValues)
-	fmt.Printf("\n")
+	ghOutputFile := os.Getenv("GITHUB_OUTPUT")
+	ghEnv, err := os.OpenFile(ghOutputFile, os.O_APPEND|os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Should this error be returned instead? Also does calling `Fatal` here prevent the file from being closed correctly? I am not sure how the `defer` is handled in this case. My concern is that the flow goes something like `log -> os.exit(1)` and then the file is never correctly closed.